### ISSUE TITLE
Select Folder GUI stays empty when disable_api_key = 1

### ISSUE THREAD

**Issue by sanderjo** [sabnzbd/sabnzbd#353]
Both on 0.8.0 Beta1 and the current git version (on Linux):

When I click on Temporary Download Folder -> Browse, I get the browser GUI popup waiting and waiting to get filled with the directories, which does not happen.

![screenshot from 2015-10-03 09 25 19](https://cloud.githubusercontent.com/assets/1273502/10261917/6e66fc68-69b1-11e5-8e96-f27073a28ccf.png)

Happens too for "Completed Download Folder" and "Watched Folder".

And that moment SAB's stdout/stderr says:

```
2015-10-03 09:33:07,151::DEBUG::[interface:439] API-call from 127.0.0.1 [Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/45.0.2454.85 Chrome/45.0.2454.85 Safari/537.36] {'output': u'json', 'apikey': u'b463b755ad289e4fd2e2e7319ab6eacf', 'mode': u'browse', 'name': u'Downloads/incomplete'}
2015-10-03 09:33:07,152::WARNING::[interface:199] Authentication missing, please enter username/password from Config->General into your 3rd party program: 127.0.0.1>Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/45.0.2454.85 Chrome/45.0.2454.85 Safari/537.36
```
and
```
2015-10-03 09:31:52,553::DEBUG::[interface:439] API-call from 127.0.0.1 [Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/45.0.2454.85 Chrome/45.0.2454.85 Safari/537.36] {'output': u'json', 'apikey': u'b463b755ad289e4fd2e2e7319ab6eacf', 'mode': u'browse', 'name': u'Downloads/complete'}
2015-10-03 09:31:52,554::WARNING::[interface:199] Authentication missing, please enter username/password from Config->General into your 3rd party program: 127.0.0.1>Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/45.0.2454.85 Chrome/45.0.2454.85 Safari/537.36
```

"Authentication missing"? How can that happen when I do something from the correctly functioning webgui?

EDIT:
Also with Firefox:

```
2015-10-03 09:40:40,026::DEBUG::[interface:439] API-call from 127.0.0.1 [Mozilla/5.0 (Windows NT 6.1; WOW64; rv:29.0) Gecko/20100101 Firefox/29.0] {'output': u'json', 'apikey': u'b463b755ad289e4fd2e2e7319ab6eacf', 'mode': u'browse', 'name': u'Downloads/incomplete'}
2015-10-03 09:40:40,026::WARNING::[interface:199] Authentication missing, please enter username/password from Config->General into your 3rd party program: 127.0.0.1>Mozilla/5.0 (Windows NT 6.1; WOW64; rv:29.0) Gecko/20100101 Firefox/29.0
```

**Comment by sanderjo** [comment]
When I remove my username/password from SAB's GUI, the problem is not there.

**Comment by shypike** [comment]
Possibly the skin does API calls instead of TAPI calls. I'll look into it.

**Comment by shypike** [comment]
Cannot reproduce on OSX or Windows.
Which version of Linux are you using?

**Comment by sanderjo** [comment]
Ah, I found the cause: disable_api_key was set to 1. And then the URL does not work ... ??!!

```
sander@flappie:~$ cat ~/.sabnzbd/sabnzbd.ini | grep disable_api_key
disable_api_key = 1

sander@flappie:~$ curl  'http://localhost:8080/sabnzbd/api?mode=browse&output=json&apikey=b463b755ad289e4fd2e2e7319ab6eacf&name=Downloads%2Fincomplete'
{"status": false, "error": "Missing authentication"}sander@flappie:~$ 
sander@flappie:~$
```

```
sander@flappie:~$ cat ~/.sabnzbd/sabnzbd.ini | grep disable_api_key
disable_api_key = 0

sander@flappie:~$ curl  'http://localhost:8080/sabnzbd/api?mode=browse&output=json&apikey=b463b755ad289e4fd2e2e7319ab6eacf&name=Downloads%2Fincomplete'
{"paths": [{"current_path": "/home/sander/Downloads/incomplete"}, {"path": "/home/sander/Downloads", "name": ".."},
```

Let's check SAB 0.7.20 ... ah, same problem: empty directory browser GUI popup, and not accepting the URL with apikey if disable_api_key = 1.

**Comment by shypike** [comment]
@thezoggy 
It looks like the file browser calls the API function instead of the TAPI.
However, I cannot find where the API is called from the templates. Is it in script.js ?

**Comment by sanderjo** [comment]
Is this it (first line):
```
$ grep -irn "mode=browse" * 
interfaces/Config/templates/_inc_header_uc.tmpl:50:                url:               '${root}api?mode=browse&output=json&apikey=$session',
interfaces/Config/templates/_inc_header_uc.tmpl:51:                autocompleteURL:   '${root}api?mode=browse&output=json&compact=1&apikey=$session'
```
Full code:
```
        \$.Browser = {
            defaults: {
                title:             'Choose Directory',
                url:               '${root}api?mode=browse&output=json&apikey=$session',
                autocompleteURL:   '${root}api?mode=browse&output=json&compact=1&apikey=$session'
            }
        };
```
If so: TAPI is better?

**Comment by sanderjo** [comment]
http://wiki.sabnzbd.org/api#toc3 says:

```
If for some reason the API key is disabled and a UI username/password is set, then you must use the ma_* parameters.
```
Ah, that looks like my case?

**Comment by shypike** [comment]
Tapi is what should have been used. Good that you found it.

**Comment by Safihre** [comment]
I checked and this is the only place still having *api* and not *tapi*

**Comment by shypike** [comment]
Solved with commit 6b547b35aed8a996f9938b254251f048f17fea4e
